### PR TITLE
EXOQuarantinePolicy: New resource

### LINKED ISSUE #1851: Add / Support EXOQuarantinePolicy Resource in Microsoft365DSC
ISSUE TITLE:
 
Add / Support **EXOQuarantinePolicy** Resource in **Microsoft365DSC**
**EXOQuarantinePolicy** is not supported / included in **Microsoft365DSC**

ISSUE DESCRIPTION (this template):

**EXOQuarantinePolicy**:
Property **'QuarantinePolicy'** should be added in class **EXOQuarantinePolicy** and should reference the **EXOMalwareFilterPolicy** property **'QuarantineTag'** as a Policy it should be nested with this policy to work as a **Required** property

**EXOQuarantinePolicy**  is momentary not supported please could you add as a new **Microsoft365DSC Resource**.

On class **EXOQuarantinePolicy** property **QuarantinePolicy** is momentary not supported could you add it as a new property.

### PR DESCRIPTION
**Pull request by malauter**
#### Pull Request (PR) description

New resource: EXOQuarantinePolicy

#### This Pull Request (PR) fixes the following issues

- Fixes #1851

### CODE REVIEW

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXOQuarantinePolicy/MSFT_EXOQuarantinePolicy.schema.mof [hunk 1]
```diff
@@ -0,0 +1,19 @@
+
+[ClassVersion("1.0.0.0"), FriendlyName("EXOQuarantinePolicy")]
+class MSFT_EXOQuarantinePolicy : OMI_BaseResource
+{
+    [Key, Description("The Identity parameter specifies the QuarantinePolicy you want to modify.")] String Identity;
+    [Write, Description("The EndUserQuarantinePermissionsValue parameter specifies the end-user permissions for the quarantine policy.")] SInt32 EndUserQuarantinePermissionsValue;
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_EXOQuarantinePolicy/MSFT_EXOQuarantinePolicy.schema.mof:6]: Looks like a typo. Did you mean to define this as Int32 (not SInt32)?

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXOQuarantinePolicy/readme.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# EXOQuarantinePolicy
+
+## Description
+
+Create or modify a new EXOQuarantinePolicy in your cloud-based organization.
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_EXOQuarantinePolicy/readme.md:5]: I would go with "Create or modify a EXOQuarantinePolicy in your cloud-based organization." (remove the 'new')

**Review by NikCharlebois — CHANGES_REQUESTED** [review]
2 items to review

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXOQuarantinePolicy/MSFT_EXOQuarantinePolicy.schema.mof (hunk 1 above)
Comment (malauter) [reply Modules/Microsoft365DSC/DSCResources/MSFT_EXOQuarantinePolicy/MSFT_EXOQuarantinePolicy.schema.mof]: fixed

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXOQuarantinePolicy/readme.md (hunk 2 above)
Comment (malauter) [reply Modules/Microsoft365DSC/DSCResources/MSFT_EXOQuarantinePolicy/readme.md]: fixed

**Review by NikCharlebois — APPROVED** [review]

**Comment by atdheekurteshi** [comment]
@NikCharlebois sorry to write a little bit late but is there an option to add also **EndUserSpamNotificationLanguage** as an attribute for **EXOQuarantinePolicy**

**Comment by malauter** [comment]
@atdheekurteshi I skipped this attribute (and some more) because they are flagged as "This parameter is reserved for internal Microsoft use." in Docs. 
@NikCharlebois is there any general advice if we should add these kind of internal parameters to our resources too or not?

**Comment by atdheekurteshi** [comment]
@malauter @NikCharlebois Since we are working for a **big client** that I can not disclose it here with more than 1000 tenants one of the attributes that we need to use is also the **EndUserSpamNotificationLanguage** so it would be nice to include it also.

**Pull request merged**